service/src/main/resources/db/migration/V281__message_thread_folder.sql

```diff
@@ -0,0 +1,19 @@
+CREATE TABLE message_thread_folder
+(
+    id                      uuid PRIMARY KEY         NOT NULL DEFAULT ext.uuid_generate_v1mc(),
+    created                 timestamp with time zone NOT NULL DEFAULT now(),
+    updated                 timestamp with time zone NOT NULL DEFAULT now(),
+    name                    text                     NOT NULL DEFAULT '',
+    owner_id                uuid                     NOT NULL REFERENCES message_account (id) ON DELETE CASCADE,
+
+    CONSTRAINT uniq$thread_folder UNIQUE (name, owner_id)
+);
+
+CREATE INDEX idx$thread_folder_owner ON message_thread_folder (id, owner_id);
+CREATE INDEX idx$thread_folder_name ON message_thread_folder (owner_id, name);
+CREATE TRIGGER set_timestamp BEFORE UPDATE ON message_thread_folder FOR EACH ROW EXECUTE PROCEDURE trigger_refresh_updated();
+
+ALTER TABLE message_thread_participant
+    ADD COLUMN folder_id uuid REFERENCES message_thread_folder (id) ON DELETE CASCADE;
+
+CREATE INDEX idx$thread_participant_folder ON message_thread_participant (participant_id, folder_id, last_message_timestamp);
```

Comment:
If you change all message thread querying to include the `folder_id` (which can be NULL), then you should modify the existing indices instead of adding this new index. The existing ones are:

```
create index idx$thread_participant_message
    on message_thread_participant (participant_id, last_message_timestamp);

create index idx$thread_participant_received
    on message_thread_participant (participant_id, last_received_timestamp)
    where (last_received_timestamp IS NOT NULL);
```

Just drop those and re-create them as `(participant_id, folder_id, last_*_timestamp)`.